array-manipulation-1/src/compact.ts

```diff
@@ -0,0 +1,12 @@
+/* exported compact */
+function compact(array: unknown[]): unknown[] {
+  const newArray = [];
+  for (let i = 0; i < array.length; i++) {
+    const checker = Boolean(array[i]);
+
+    if (checker !== false) {
+      newArray.push(array[i]);
+    }
+  }
+  return newArray;
+}
```

Comment:
While this looks like it works, there is a better way of determining whether or not you want to show/remove the item.

What you are doing is using the `Boolean` object that represents a boolean and passing it a value. This coerces a _truthy_ or a _falsy_ value to `true` or `false`.

Instead of passing your `array[i]` to the `Boolean` object, what happens if you just pass it through to your `if` statement's conditional expression? 😉